### PR TITLE
feat: add jupiter price client

### PR DESCRIPTION
**Pull request by rhbrunetto**
## Changes

- Adds Jupiter Price Client to `jupiter_aggregator` package

## Related issues

Breakdown of #632 

## Checklist

- [x] PR is ready for review (if not, it should be a draft).
- [x] PR title follows [Conventional Commits][1] guidelines.
- [ ] Screenshots/video added.
- [ ] Tests added.
- [x] Self-review done.

[1]: https://www.conventionalcommits.org/en/v1.0.0/

### CODE REVIEW

File: packages/cryptoplease_link/jupiter_aggregator/lib/src/client.dart [hunk 1]
```diff
@@ -24,7 +26,30 @@ abstract class JupiterAggregatorClient {
...
+  );
+}
+
+/// For docs head to https://docs.jup.ag/jupiter-api/price-api-for-solana
+@RestApi(baseUrl: 'https://price.jup.ag/v1')
+abstract class JupiterPriceClient {
+  factory JupiterPriceClient() => _JupiterPriceClient(
+        Dio()
+          ..interceptors.add(
+            InterceptorsWrapper(
```
Comment (ookami-kb) [inline packages/cryptoplease_link/jupiter_aggregator/lib/src/client.dart:40]: Why do you need this interceptor?

File: packages/cryptoplease_link/jupiter_aggregator/lib/src/model.dart [hunk 2]
```diff
@@ -150,15 +138,50 @@ class QuoteRequestDto with _$QuoteRequestDto {
...
+  factory PriceRequestDto.fromJson(Map<String, dynamic> json) =>
+      _$PriceRequestDtoFromJson(json);
+}
 
-  factory SwapRequestDto.fromJson(Map<String, dynamic> json) =>
-      _$SwapRequestDtoFromJson(json);
+@freezed
+class PriceResponseDto with _$PriceResponseDto {
+  const factory PriceResponseDto({
+    @JsonKey(name: 'data', readValue: _readPrice) required double price,
```
Comment (ookami-kb) [inline packages/cryptoplease_link/jupiter_aggregator/lib/src/model.dart:179]: This is very confusing. Make it closer to the real structure, i.e. add `Data(double price)` intermediate DTO.

**Review by ookami-kb — CHANGES_REQUESTED** [review]

File: packages/cryptoplease_link/jupiter_aggregator/lib/src/client.dart (hunk 1 above)
Comment (rhbrunetto) [reply packages/cryptoplease_link/jupiter_aggregator/lib/src/client.dart:40]: Jupiter Price has not a content-type header in the response, so `Dio` cannot infer that the payload is a Json, resulting in a type error when calling `PriceResponseDto.fromJson`. Seems like `retrofit` team won't address this issue and that's the solution I found out:  https://github.com/trevorwang/retrofit.dart/issues/276

File: packages/cryptoplease_link/jupiter_aggregator/lib/src/model.dart [hunk 3]
```diff
@@ -150,15 +138,57 @@ class QuoteRequestDto with _$QuoteRequestDto {
...
+  }) = _PriceRequestDto;
+
+  factory PriceRequestDto.fromJson(Map<String, dynamic> json) =>
+      _$PriceRequestDtoFromJson(json);
+}
+
+@freezed
+class PriceResponseDto with _$PriceResponseDto {
+  const factory PriceResponseDto({
+    @JsonKey(name: 'data') required PriceDto data,
```
Comment (ookami-kb) [inline packages/cryptoplease_link/jupiter_aggregator/lib/src/model.dart:179]:
```suggestion
    required PriceDto data,
```

**Review by ookami-kb — APPROVED** [review]

**Pull request merged**